File: Backend/Models/ProductModel/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace App.Models.ProductModel
{
      public class Product
      {
            public int Id { get; set; }
            [Required]
            public string Name { get; set; } = string.Empty;
            public string NormalizedName { set; get; } = string.Empty;
            public string Description { get; set; } = string.Empty;
            public string Thumbnail { get; set; }
            public string[] ImagePaths { get; set; }
            public string[] Tags { get; set; }
            [Display(Name = "Url")]
            public string Slug { get; set; }
            public int Price { get; set; }
            public int SalePrice { get; set; }
            public DateTime CreateAt { get; set; }
            public virtual List<ProductCategory> ProductCategories { get; set; }
            public int IconId { get; set; }
            [ForeignKey("IconId")]
            public Icon Icon { get; set; }
            public DateTime UpdateAt { get; set; }
            public string Technology { get; set; }
      }
}

File: frontend/service.store/src/models/product.ts
import { Icon } from "./icon"
import { Category } from "./category"

export interface Product {
    id: string | number
    name: string
    category: Category | null
    brand: Icon | null
    description: string | null
    price: number | null
    salePrice?: number | null
    imagePaths: string[]
    thumbnail: string
    tags?: string[]
    detail?: string
}

export type CartItemReq = {
    productId: number
    quantity: number
}

export type CartItem = {
    quantity: number
    productId: number
}

export type CartRes = {
    id: number | string
    items: CartItem[]
}
